text/0202-events-and-logs-api.md

```diff
@@ -0,0 +1,172 @@
+
+Note: This could, for example, be implemented by storing any mutable configuration in the LoggerProvider and having Logger implementation objects have a reference to the LoggerProvider from which they were obtained. If configuration must be stored per-Logger (such as disabling a certain Logger), the Logger could, for example, do a look-up with its name+version+schema_url+event_domain+include_trace_context in a map in the LoggerProvider, or the LoggerProvider could maintain a registry of all returned Loggers and actively update their configuration if it changes.
+
+The effect of associating a Schema URL with a Logger MUST be that the telemetry emitted using the Logger will be associated with the Schema URL, provided that the emitted data format is capable of representing such association.
+
+### Logger
+
+The Logger is responsible for creating Events and Logs.
+
+Note that Loggers should not be responsible for configuration. This should be the responsibility of the LoggerProvider instead.
+
+#### Logger operations
+
+The Logger MUST provide functions to:
+
+- Create an `Event` and emit it to the processing pipeline.
+  - The API MUST accept an event name as a parameter. The event name provided should be inserted as an attribute with key `event.name`. Care MUST be taken to not override or delete this attribute while the `Event` is created.  This function MAY be named `logEvent`.
+- Create a `Log Record` and emit it to the processing pipeline.
+  - This function MAY be named `logRecord`.
+  - The intended users of this API is Log Appenders.
+
+### Usage
+
+```java
+OpenTelemetry openTelemetry = OpenTelemetry.noop();
+Logger logger = openTelemetry.getLogger("my-scope");
+
+// Using the convenience method to log an event directly
+logger.logEvent("network-changed", 
+                 Attributes.builder().put("type", "wifi").build());
+
+// Using the event builder to log an event
+logger.eventBuilder("page-navigated").build().setAttribute("url", "http://foo.com/bar#new").emit();
+
+// Using the logRecord builder to log a record
+Logger logger = openTelemetry.getLogger("another-scope");
+logger.logRecordBuilder().build().setBody("I am a log message").emit();
+
+```
+
+### Usage in Client-side telemetry
+
+```java
+public void addBrowserEvent(String name, Attributes attributes) {
+   Logger logger = openTelemetry.getLogger("my-scope", "1.0", "browser");
+   logger.logEvent(name, attributes);
+}
+
+public void addMobileEvent(String name, Attributes attributes) {
+   Logger logger = openTelemetry.getLogger("my-scope", "1.0", "mobile");
+   logger.logEvent(name, attributes);
+}
+```
+
+### Usage in eBPF
+
+From the eBPF [demo](https://youtu.be/F1VTRqEC8Ng?t=233), it looks like they have chosen to put the Event data in the Body field of LogRecord instead of Attributes.  They might have to make the changes to move the event data to attributes to conform to this API specification.
+
+## Semantic Convention for event attributes
```

Comment:
This section is extremely specific. At the risk of repeating myself, this reads more like a draft spec than an OTEP. I see rigid definitions but no arguments for the value or usefulness of any of the specified features/requirements.

Reply:
> I would almost argue for the removal of the `Events and Logs API Interface` section here and replacing it with use cases and motivations. I see very little about what these APIs are expected to be used for, alternatives that were considered, and why they are not sufficient.

@dyladan I see your point and can add more content in the OTEP on use-cases and motivations. It was discussed in the calls but if you feel it is not represented enough here I will address that.

Reply:
@dyladan I added some text in this commit to address some of what you asked but it's not elaborate. The primary motivation of this OTEP is really to distinguish between Events and Logs, both represented using the `LogRecord` data model, via semantic conventions and the API is to enforce those conventions - that's why there's not really much to elaborate on use-cases and motivation. So far what we agreed on during the discussion on this OTEP is having an `event.name` attribute to identify a `LogRecord` as being an `Event`, and that the event data is mostly in the attributes.

Reply:
@dyladan Regarding the inclusion of `Events and Logs API Interface` section in this OTEP, we did that only to keep things together for easier understanding of the contents. Apparently, few other OTEPs have done this in the past too. Let me know if you feel strongly about its removal from the OTEP.

Reply:
Removed the API interface section.